pkg/api/v20170701/types.go

```diff
@@ -0,0 +1,294 @@
+	// ApiServerCertificate is the rest api server certificate, and signed by the CA
+	APIServerCertificate string `json:"apiServerCertificate,omitempty"`
+	// ApiServerPrivateKey is the rest api server private key, and signed by the CA
+	APIServerPrivateKey string `json:"apiServerPrivateKey,omitempty"`
+	// ClientCertificate is the certificate used by the client kubelet services and signed by the CA
+	ClientCertificate string `json:"clientCertificate,omitempty"`
+	// ClientPrivateKey is the private key used by the client kubelet services and signed by the CA
+	ClientPrivateKey string `json:"clientPrivateKey,omitempty"`
+	// KubeConfigCertificate is the client certificate used for kubectl cli and signed by the CA
+	KubeConfigCertificate string `json:"kubeConfigCertificate,omitempty"`
+	// KubeConfigPrivateKey is the client private key used for kubectl cli and signed by the CA
+	KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
+	// caPrivateKey is an internal field only set if generation required
+	caPrivateKey string
+}
+
+// LinuxProfile represents the Linux configuration passed to the cluster
+type LinuxProfile struct {
+	AdminUsername string `json:"adminUsername"`
+
+	SSH struct {
+		PublicKeys []struct {
+			KeyData string `json:"keyData"`
+		} `json:"publicKeys"`
+	} `json:"ssh"`
+	Secrets []KeyVaultSecrets `json:"secrets,omitempty"`
+}
+
+// WindowsProfile represents the Windows configuration passed to the cluster
+type WindowsProfile struct {
+	AdminUsername string            `json:"adminUsername,omitempty"`
+	AdminPassword string            `json:"adminPassword,omitempty"`
+	Secrets       []KeyVaultSecrets `json:"secrets,omitempty"`
+}
+
+// ProvisioningState represents the current state of container service resource.
+type ProvisioningState string
+
+const (
+	// Creating means ContainerService resource is being created.
+	Creating ProvisioningState = "Creating"
+	// Updating means an existing ContainerService resource is being updated
+	Updating ProvisioningState = "Updating"
+	// Failed means resource is in failed state
+	Failed ProvisioningState = "Failed"
+	// Succeeded means resource created succeeded during last create/update
+	Succeeded ProvisioningState = "Succeeded"
+	// Deleting means resource is in the process of being deleted
+	Deleting ProvisioningState = "Deleting"
+	// Migrating means resource is being migrated from one subscription or
+	// resource group to another
+	Migrating ProvisioningState = "Migrating"
+)
+
+// OrchestratorProfile contains Orchestrator properties
+type OrchestratorProfile struct {
+	OrchestratorType    OrchestratorType    `json:"orchestratorType"`
+	OrchestratorVersion OrchestratorVersion `json:"orchestratorVersion"`
+}
+
+// MasterProfile represents the definition of master cluster
+type MasterProfile struct {
+	Count                    int    `json:"count"`
+	DNSPrefix                string `json:"dnsPrefix"`
+	VMSize                   string `json:"vmSize"`
```

Comment:
Ok, thanks. Looks like I might have read an older version of the description. You should add FirstConsecutiveStaticIP to the description as well since its being exposed in the API.